src/vcpkg/commands.update-baseline.cpp

```diff
@@ -0,0 +1,192 @@
+            }
+            else
+            {
+                msg::println(msgUpdateBaselineUpdatedBaseline,
+                             msg::url = url,
+                             msg::old_value = old_baseline,
+                             msg::new_value = *p);
+            }
+        }
+        else
+        {
+            msg::println(msgUpdateBaselineNoUpdate, msg::url = url, msg::value = old_baseline);
+        }
+    }
+
+    void UpdateBaselineCommand::perform_and_exit(const VcpkgCmdArguments& args, const VcpkgPaths& paths) const
+    {
+        auto options = args.parse_arguments(COMMAND_STRUCTURE);
+
+        const bool add_builtin_baseline = Util::Sets::contains(options.switches, OPTION_ADD_INITIAL_BASELINE);
+        const bool dry_run = Util::Sets::contains(options.switches, OPTION_DRY_RUN);
+
+        auto configuration = paths.get_configuration();
+        const bool has_manifest = paths.get_manifest().has_value();
+        auto manifest = has_manifest ? *paths.get_manifest().get() : ManifestAndLocation{};
+
+        if (configuration.location == ConfigurationLocation::None && !has_manifest)
+        {
+            msg::print_warning(msgUpdateBaselineNoConfiguration);
+            Checks::exit_success(VCPKG_LINE_INFO);
+        }
+
+        bool has_builtin_baseline = manifest.manifest.contains("builtin-baseline");
+
+        if (add_builtin_baseline && !has_manifest)
+        {
+            Checks::msg_exit_with_error(
+                VCPKG_LINE_INFO, msgUpdateBaselineAddBaselineNoManifest, msg::option = OPTION_ADD_INITIAL_BASELINE);
+        }
+        if (!has_builtin_baseline && !add_builtin_baseline && configuration.location == ConfigurationLocation::None)
+        {
+            msg::print_warning(msgUpdateBaselineNoExistingBuiltinBaseline, msg::option = OPTION_ADD_INITIAL_BASELINE);
+            Checks::exit_success(VCPKG_LINE_INFO);
+        }
+
+        LocalizedString error;
+
+        if (has_builtin_baseline || add_builtin_baseline)
+        {
+            // remove default_reg, since that's filled in with the builtin-baseline
+            configuration.config.default_reg = nullopt;
+
+            auto synthesized_registry = RegistryConfig{};
+            synthesized_registry.kind = "builtin";
+
+            auto result = synthesized_registry.get_latest_baseline(paths, error);
+
+            auto p_builtin_baseline = manifest.manifest.get("builtin-baseline");
+            auto old_baseline = p_builtin_baseline ? Optional<StringView>(p_builtin_baseline->string()) : nullopt;
+            print_update_message(synthesized_registry.registry_location(), old_baseline, result);
+
+            if (auto p = result.get())
+            {
+                manifest.manifest.insert_or_replace("builtin-baseline", std::move(*p));
+            }
+            else if (!error.empty())
+            {
+                msg::print_warning(error);
```

Comment:
Is it a warning or an error?

Reply:
it is a warning, because an error for _this_ function ends up just being a warning.